spec/models/green_lanes/goods_nomenclature_spec.rb

```diff
@@ -14,4 +14,46 @@
 
     it { is_expected.to include :applicable_category_assessments }
   end
+
+  describe '#primary_assessments_group' do
+    subject { goods_nomenclature.primary_assessments_group }
+
+    let :goods_nomenclature do
+      build :green_lanes_goods_nomenclature,
+            applicable_category_assessments: assessments
+    end
+
+    context 'with only category 1' do
+      let(:assessments) { attributes_for_pair :category_assessment, category: 1 }
+
+      it { is_expected.to have_attributes length: 2 }
+      it { is_expected.to all have_attributes category: 1 }
+    end
+
+    context 'with only category 2' do
+      let(:assessments) { attributes_for_pair :category_assessment, category: 2 }
+
+      it { is_expected.to have_attributes length: 2 }
+      it { is_expected.to all have_attributes category: 2 }
+    end
+
+    context 'with mix of category 1 and category 2' do
```

Comment:
context "with both categories: 1 and 2"

Reply:
Good call - amended the context labels for consistency